### PR TITLE
TST: show_versions test with unmerged commits

### LINKED ISSUE #32120: test_show_versions failing locally
```
>       assert re.search(r"commit\s*:\s[0-9a-f]{40}\n", result)
E       AssertionError: assert None
E        +  where None = <function search at 0x102703560>('commit\\s*:\\s[0-9a-f]{40}\\n', '\nINSTALLED VERSIONS\n------------------\ncommit           : None\npython           : 3.7.6.final.0\npython-bits     ...   : 0.14.1\nxlrd             : 1.2.0\nxlwt             : 1.3.0\nxlsxwriter       : 1.2.1\nnumba            : 0.47.0\n')
E        +    where <function search at 0x102703560> = re.search
```

### PR DESCRIPTION
**Pull request by jbrockmendel**
- [x] closes #32120
- [x] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

### CODE REVIEW

**Comment by jbrockmendel** [comment]
Closing pending investigation in #32120.

**Comment by jbrockmendel** [comment]
Re-opened, the issue wasnt with un-merged commits, but with git worktree.  re-using versioneer code handles the problem.

**Comment by jbrockmendel** [comment]
gentle ping @simonjayhawkins @WillAyd ; this is workflow-affecting

**Review by simonjayhawkins — APPROVED** [review]
@jbrockmendel lgtm

**Review by simonjayhawkins — APPROVED** [review]
@jbrockmendel lgtm

File: pandas/tests/util/test_show_versions.py [hunk 1]
```diff
@@ -1,8 +1,26 @@
 import re
 
+import pytest
+
 import pandas as pd
 
 
+@pytest.mark.filterwarnings(
```
Comment (WillAyd) [inline pandas/tests/util/test_show_versions.py]: Are these actually required as part of this change? The linkage here with the rest of the PR isn’t totally clear

**Review by WillAyd — COMMENTED** [review]
Quick question but generally lgtm

File: pandas/tests/util/test_show_versions.py (hunk 1 above)
Comment (jbrockmendel) [reply pandas/tests/util/test_show_versions.py]: these are just because its annoying to have in the output

**Comment by WillAyd** [comment]
OK thanks @jbrockmendel 

Heads up - I’m at CZI this week so may be slow to respond to some of your other PRs. Should catch up over the weekend at the latest

**Pull request merged**